watchdog_reader.go

```diff
@@ -23,9 +25,14 @@ func newWatchdogReader(reader io.Reader, timeout time.Duration, timer *time.Time
 
 // Read reads up to len(p) bytes into p
 func (t *watchdogReader) Read(p []byte) (n int, err error) {
-	// FIXME limit the amount of data read in one chunk so as to not exceed the timeout?
+	//never read more bytes than watchdogChunkSize
+	readTarget := p
+	if len(p) > watchdogChunkSize {
+		readTarget = p[0:watchdogChunkSize]
+	}
+
```

Comment:
I don't really see the advantage, since it won't reduce the amount of timer resets or actual reads, and `io.Reader.Read` already demands that callers need to be prepared to handle a partial read, but okay...